File: Java Fundamentals-Jan2020/src/b_BasicSyntaxCondStatementsLoops_Exercise/VendingMachine.java
package b_BasicSyntaxCondStatementsLoops_Exercise;

import java.util.Scanner;

public class VendingMachine {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String command = "";

        double money = 0;

        while (true) {
            command = sc.nextLine();

            if (command.equals("Start")) {
                break;
            }

            double coin = Double.parseDouble(command);
            if (coin == 0.1 || coin == 0.2 || coin == 0.5 || coin == 1 || coin == 2) {
                money += coin;
            } else {
                System.out.println(String.format("Cannot accept %.2f", coin));
            }
        }

        while (true) {
            command = sc.nextLine();

            double price = 0;

            switch (command) {
                case "Nuts": price = 2; break;
                case "Water": price = 0.7; break;
                case "Crisps": price = 1.5; break;
                case "Soda": price = 0.8; break;
                case "Coke": price = 1; break;
                case "End": break;
                default:
                    System.out.println("Invalid product");
                    break;
            }

            if (command.equals("End")) {
                System.out.println(String.format("Change: %.2f", money));
                break;
            }

            if (money < price) {
                System.out.println("Sorry, not enough money");
            } else if (money >= price && price != 0) {
                System.out.println(String.format("Purchased %s", command));
                money -= price;
            }
        }
    }
}

File: CSharp-Fundamentals-2023/01-1.ConditStatementsAndLoops-Exercise/07.VendingMachine/VendingMachine.cs
// Accumulate money by inserting valid coins
double money = 0.0;string inputCoin = Console.ReadLine();while (inputCoin != "Start")
{
    double coin = double.Parse(inputCoin);

    //Check for valid coin
    if (coin == 0.1 || coin == 0.2 || coin == 0.5 || coin == 1 || coin == 2)
    {
        money += coin;
    }
    else
    {
        Console.WriteLine($"Cannot accept {coin}");
    }

    inputCoin = Console.ReadLine();
}


// Buy valid products
// • "Nuts" with a price of 2.0
// • "Water" with a price of 0.7
// • "Crisps" with a price of 1.5
// • "Soda" with a price of 0.8
// • "Coke" with a price of 1.0

string inputProduct = Console.ReadLine();

while (inputProduct != "End")
{
    double productPrice = 0;
    switch (inputProduct)
    {
        case "Nuts": productPrice = 2.0; break;
        case "Water": productPrice = 0.7; break;
        case "Crisps": productPrice = 1.5; break;
        case "Soda": productPrice = 0.8; break;
        case "Coke": productPrice = 1.0; break;
        default: 
            Console.WriteLine("Invalid product");
            inputProduct = Console.ReadLine();
            continue;
    }

    if (money - productPrice >= 0)
    {
        Console.WriteLine($"Purchased {inputProduct.ToLower()}");
        money -= productPrice;
    }
    else
    {
        Console.WriteLine("Sorry, not enough money");
    }

    inputProduct = Console.ReadLine();
}

Console.WriteLine($"Change: {money:f2}");